src/tyr/matrix_serializer.cc

```diff
@@ -43,6 +43,32 @@ json::ArrayPtr serialize_distance(const valhalla::Matrix& matrix,
   }
   return distance;
 }
+
+json::ArrayPtr serialize_shape(const valhalla::Matrix& matrix,
+                               const size_t start_td,
+                               const size_t td_count,
+                               const ShapeFormat shape_format) {
+  // TODO(nils): shapes aren't implemented yet in TDMatrix
+  auto shapes = json::array({});
+  if (shape_format == no_shape)
+    return shapes;
+
+  for (size_t i = start_td; i < start_td + td_count; ++i) {
+    switch (shape_format) {
+      // even if it source == target or no route found, we want to emplace an element
+      case geojson:
+        if (!matrix.shapes()[i].empty())
+          shapes->emplace_back(tyr::geojson_shape(decode<std::vector<PointLL>>(matrix.shapes()[i])));
+        else
+          shapes->emplace_back(json::map({}));
```

Comment:
why not `null` i think its better than empty map:

```suggestion
          shapes->emplace_back(nullptr);
```

Reply:
yeah let's do that